### ISSUE TITLE
save SCTransform with different parameters in the same Seurat object

### ISSUE THREAD

**Issue by yeroslaviz** [satijalab/seurat#3356]
Is it possible to run `SCTransform` with different parameters on the same Seurat Object and save them all within the same object?

I tried something like this

```
seurat_filtered.sct <- SCTransform(seurat_filtered, vars.to.regress = "mitoRatio",
                                   assay = "RNA", new.assay.name = "SCT_varMT", verbose = FALSE)

seurat_filtered.sct <- SCTransform(seurat_filtered, vars.to.regress = c("mitoRatio", "sample"), 
                                   assay = "RNA", new.assay.name = "SCT_varMT_Samples", verbose = FALSE)
```
but each time only the default assay `RNA` and only one more assay is saved.

E.g. 

```
> seurat_filtered.sct
An object of class Seurat 
44170 features across 76412 samples within 2 assays 
Active assay: SCT_varMT (22085 features, 3000 variable features)
 1 other assay present: RNA
```

How can I save both in the seurat object?

thanks

**Comment by timoast** [comment]
Here you're running SCTransform a second time using the original Seurat object, not the object where you've SCTransform the first time.

```r
seurat_filtered <- SCTransform(seurat_filtered, vars.to.regress = "mitoRatio",
                                   assay = "RNA", new.assay.name = "SCT_varMT", verbose = FALSE)

seurat_filtered <- SCTransform(seurat_filtered, vars.to.regress = c("mitoRatio", "sample"), 
                                   assay = "RNA", new.assay.name = "SCT_varMT_Samples", verbose = FALSE)
```

**Comment by yeroslaviz** [comment]
This i the same object.  In the first run I used the filtered object to run the transformation based on only the mito. ratio. The results I saved in the same object in a second assay named `SCT_varMT`. I then ran a second analysis with different regression parameters, but wanted to save them to the same Seurat object.

I was expecting Seurat to create three different assays within the `seurat_filtered` object with the names `RNA`, `SCT_varMT` and `SCT_varMT_Samples`. But it doesn't. 

Is it not possible to do so?
